src/main/java/se/liu/ida/hefquin/engine/query/impl/QueryPatternUtils.java

```diff
@@ -170,18 +171,18 @@ public static int getNumberOfBNodeOccurrences( final BGP bgp ) {
 	 * Returns the number of occurrences of variables in the given graph
 	 * pattern. If the same variable occurs multiple times, each occurrence
 	 * is counted.
-	 * 
+	 *
 	 * If the given pattern is a {@link TriplePattern}, this function returns
 	 * the result of {@link #getNumberOfVarOccurrences(TriplePattern)}.
 	 * Similarly, if the given pattern is a {@link BGP}, this function
 	 * returns the result of {@link #getNumberOfVarOccurrences(BGP)}.
 	 */
 	public static int getNumberOfVarOccurrences( final SPARQLGraphPattern queryPattern ) {
-		if ( queryPattern instanceof TriplePattern ) {
-			return getNumberOfVarOccurrences( (TriplePattern) queryPattern );
+		if ( queryPattern.asJenaOp() instanceof Triple) {
+			return getNumberOfVarOccurrences( new TriplePatternImpl( (Triple) queryPattern.asJenaOp()) );
 		}
-		else if ( queryPattern instanceof BGP ) {
-			return getNumberOfVarOccurrences( (BGP) queryPattern );
+		else if ( queryPattern.asJenaOp() instanceof OpBGP) {
+			return getNumberOfVarOccurrences( createBGP(((OpBGP) queryPattern.asJenaOp()).getPattern()) );
```

Comment:
Same comment here. Do not create a new `BGP` object only for the purpose of calling this method.